### PR TITLE
Switch to Sidekiq.

### PR DESCRIPTION
**Pull request by justinlittman**
## Why was this change made? 🤔
Surely not everyone could be kung fu fighting?

## How was this change tested? 🤨

⚡ ⚠ If this change involves consuming from other services or writing to shared file systems, test that web archive seed and crawl accessioning works properly in [stage|qa] environment, in addition to specs. ⚡

### CODE REVIEW

File: lib/dor/was_seed/desc_metadata_generator.rb [hunk 1]
```diff
@@ -5,8 +5,8 @@ module WasSeed
     class DescMetadataGenerator < MetadataGenerator
       attr_reader :desc_metadata_name, :seed_uri, :collection_id
 
-      def initialize(workspace_path, druid_id, seed_uri, collection_id)
-        super(workspace_path, druid_id)
+      def initialize(workspace_path, druid_id, seed_uri, collection_id, logger: nil)
```
Comment (ndushay) [inline lib/dor/was_seed/desc_metadata_generator.rb:8]: was robots need their own logger?

File: lib/robots/dor_repo/was_crawl_dissemination/cdxj_generator.rb [hunk 2]
```diff
@@ -8,11 +8,7 @@ def initialize
           super('wasCrawlDisseminationWF', 'cdxj-generator')
         end
 
-        # `perform` is the main entry point for the robot. This is where
```
Comment (ndushay) [inline lib/robots/dor_repo/was_crawl_dissemination/cdxj_generator.rb:11]: curiously - this context almost helps me, while the  `super()` line in constructor with line comment feels superfluous.  But I can live without this comment.   

Maybe someday we can rid ourselves of these centralized/decentralized robots.

File: spec/robots/dor_repo/was_crawl_preassembly/build_was_crawl_druid_tree_spec.rb [hunk 3]
```diff
@@ -22,7 +22,7 @@
     # let(:version_client) { instance_double(Dor::Services::Client::ObjectVersion, current: '1') }
 
     before do
-      # allow(WorkflowClientFactory).to receive(:build).and_return(workflow_client)
+      # allow(LyberCore::WorkflowClientFactory).to receive(:build).and_return(workflow_client)
```
Comment (ndushay) [inline spec/robots/dor_repo/was_crawl_preassembly/build_was_crawl_druid_tree_spec.rb:25]: i know, global search and replace ... but ...

File: spec/robots/dor_repo/was_seed_preassembly/thumbnail_generator_spec.rb [hunk 4]
```diff
@@ -69,8 +69,8 @@
       end
 
       it 'raises an error' do
-        # instance.perform(druid)
-        expect { instance.perform(druid) }.to raise_error(StandardError)
+        # test_perform(instance, druid)
```
Comment (ndushay) [inline spec/robots/dor_repo/was_seed_preassembly/thumbnail_generator_spec.rb:72]: this comment can be removed, IMO

**Review by ndushay — APPROVED** [review]
a few inline quibbles for your consideration.

**Comment by ndushay** [comment]
... "HA!"  in response to your description.

File: spec/spec_helper.rb [hunk 5]
```diff
@@ -11,14 +8,17 @@
   add_filter '/vendor/'
 end
 
+ENV['ROBOT_ENVIRONMENT'] = 'test'
```
Comment (ndushay) [inline spec/spec_helper.rb:11]: do we still need this, if we got rid of the config/environments?

File: lib/dor/was_seed/desc_metadata_generator.rb (hunk 1 above)
Comment (justinlittman) [reply lib/dor/was_seed/desc_metadata_generator.rb:8]: The old lyber-core used a global logger. The new lyber-core uses Sidekiq's logger, which must be passed around.

File: spec/robots/dor_repo/was_crawl_preassembly/build_was_crawl_druid_tree_spec.rb (hunk 3 above)
Comment (justinlittman) [reply spec/robots/dor_repo/was_crawl_preassembly/build_was_crawl_druid_tree_spec.rb:25]: De-crufted.

File: spec/robots/dor_repo/was_seed_preassembly/thumbnail_generator_spec.rb (hunk 4 above)
Comment (justinlittman) [reply spec/robots/dor_repo/was_seed_preassembly/thumbnail_generator_spec.rb:72]: Removed.

**Pull request merged**